src/celeritas/user/ActionDiagnostic.cc

```diff
@@ -218,16 +220,24 @@ void ActionDiagnostic::clear()
  */
 void ActionDiagnostic::begin_run_impl(CoreParams const& params)
 {
-    CELER_EXPECT(!store_);
+    if (!*this)
+    {
+        static std::mutex initialize_mutex;
+        std::lock_guard<std::mutex> scoped_lock{initialize_mutex};
```

Comment:
OpenMP doesn't make guarantees about std threading working with it... maybe we should leave this as-is but break https://github.com/celeritas-project/celeritas/issues/418#issuecomment-1358063353 into a separate follow-on issue?

Reply:
Yeah, I was wondering about this... your comment sounds like a good plan.